### PR TITLE
[DEV-2674] - Adding new protocol to MemoryAgingStore

### PR DESCRIPTION
**Pull request by drew-patel**
Created a new protocol which will allow the application to modify a user's session who is not controlling the application.

### CODE REVIEW

**Comment by drew-patel** [comment]
If you look at this code it is what I **would** need to do to get the protocol to work (if the `aging-session` code sat inside de.web). Since the protocol and some other code we need are outside of the `aging-session` library, I'm not sure how to implement it. 

If I move the new protocol and functions into this library, it may work but then we are exposing internal code from this public library.

Please let me know your thoughts.

**Comment by e45lee** [comment]
Unfortunately @Elknar is on vacation this week and next...

**Comment by Elknar** [comment]
as an option, we could consider providing all the sente-specific fns as arguments to the kill-user-session

although it would be ugly... in any case, lets go over this in-person as agreed on slack

File: src/aging_session/modify_user_session.clj [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(ns aging-session.modify-user-session)
+
+(defprotocol ModifyUserSession
+  "A protocol for modifying the session(s) of a user who is not currently logged in and
+   controlling the application."
+
+  (terminate-user-session
+    [store x]
+    [store x y]
```
Comment (e45lee) [inline src/aging_session/modify_user_session.clj]: Just do (terminate-user-session session-key), perhaps, and pull out the logic for transforming user-id + firm-id -> session key into another protocol (could even do that in de.web TBH)

Comment (drew-patel) [reply src/aging_session/modify_user_session.clj]: Cool. I will be doing as discussed. Thanks!

**Review by Elknar — APPROVED** [review]

**Pull request merged**